networking_dvs/plugins/ml2/drivers/mech_dvs/agent/vcenter_util.py

```diff
@@ -298,6 +310,21 @@ def _run(self):
                                 self._handle_virtual_machine(update, now)
 
                 for port_desc in self.changed:
+                    ports_by_mac = defaultdict(dict)
+
+                    port = self.vc_api.mac_port_map.get(port_desc.mac_address, {})
+                    port.update({
+                        'port_desc': port_desc,
+                        'port': {
+                            'binding:vif_details': {
+                                'dvs_port_key': port_desc.port_key,
+                                'dvs_uuid': port_desc.dvs_uuid
+                            }, 'mac_address': port_desc.mac_address}
+                    })
+
+                    ports_by_mac[port_desc.mac_address] = port
+                    port_desc.dvs_thread = eventlet.spawn(self.vc_api.read_dvs_ports, ports_by_mac)
```

Comment:
Here you do a read_dvs_port request per port. It might be better to assemble ports_by_mac over all objects in self.changed, and then query the whole batch.

But that requires some performance testing, what is actually the better solution.

One additional point: The `self.changed` contains also deleted ports, see https://github.com/mmidolesov/networking-dvs/blob/2fe1313aa229d074ee10ff9284f086a121b51771/networking_dvs/plugins/ml2/drivers/mech_dvs/agent/vcenter_util.py#L400

Those should not be queried, and the dvs_thread condition probably should be adapted accordingly.